src/app/ApplicationTemplate.Presentation/Framework/Review/ReviewPromptService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ApplicationTemplate;
+
+/// <summary>
+/// Implementation of <see cref="IReviewPromptService"/>.
+/// </summary>
+public sealed class ReviewPromptService : IReviewPromptService
```

Comment:
You probably want this class and the interface to be generic so that you can implement tracking of custom data.

Reply:
It's not the only way, but we need to think of ways to support more than just the default ReviewSettings.

Reply:
Updated the service with an update method which can be used by tracking methods. We have default tracking methods in the service and if someone need something custom they can always create an extension and call the update method.